### DISCUSSION TITLE
How to reload cached value?

### DISCUSSION THREAD

**Discussion by mfonnesbeck** [App-vNext/Polly#2415] [Q&A]
We are caching some accounting items in memory in our web application, since they don't change often, however, we've run into a situation when we are setting up the accounting for a tenant to our associated accounting software it caches some items before they are fully setup and then we can't see all the updates to the setup process because the cache is holding onto old stored values. I see that there is no way to remove a stored value from the cache by key, and I have seen some use of a cancellation token in the documentation, but it is not fully explained how to use it. I am looking for any way to reload a cached value or remove so that I can re-add a cached value. We are using Polly 7.1.1 and Polly.Caching.Memory 3.0.2.

**Comment by peter-csala** [comment]
I haven't used the Cache policy in the past years but if my memory serves well you can access the underlying Cache provider (either `IAsyncCacheProvider` or `ISyncCacheProvider`). Both interfaces defines a `Put{Async}` method which works as an upsert. I would start the investigation there.
